src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-date-modal/individual-extension-date-modal.component.ts

```diff
@@ -0,0 +1,143 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import moment from 'moment-timezone';
+import { DateTimeHelper } from '../../../../services/datetime-helper';
+import { SimpleModalService } from '../../../../services/simple-modal.service';
+import { TimezoneService } from '../../../../services/timezone.service';
+import { DateFormat } from '../../../components/datepicker/datepicker.component';
+import { SimpleModalType } from '../../../components/simple-modal/simple-modal-type';
+import { TimeFormat } from '../../../components/timepicker/timepicker.component';
+
+export enum RadioOptions {
+  EXTEND_TO = 1,
+  EXTEND_BY = 2,
+}
+
+/**
+ * Modal to confirm permanent deletion of a feedback session.
+ */
+@Component({
+  selector: 'tm-individual-extension-date-modal',
+  templateUrl: './individual-extension-date-modal.component.html',
+  styleUrls: ['./individual-extension-date-modal.component.scss'],
+})
+
+export class IndividualExtensionDateModalComponent {
+  @Input()
+  numberOfStudents: number = 0;
+
+  @Input()
+  numberOfInstructors: number = 0;
+
+  @Input()
+  feedbackSessionEndingTime: number = 0;
+
+  @Input()
+  feedbackSessionTimeZone: string = '';
+
+  @Output() onConfirmCallBack: EventEmitter<number> = new EventEmitter();
+
+  constructor(public activeModal: NgbActiveModal,
+              private timeZoneService: TimezoneService,
+              private simpleModalService: SimpleModalService) {}
+
+  RadioOptions: typeof RadioOptions = RadioOptions;
+  radioOption: RadioOptions = RadioOptions.EXTEND_BY;
+  extendByDeadlineKey: String = '';
+  extendByDeadlineOptions: Map<String, Number> = new Map([
+    ['12 hours', 0.5],
+    ['1 day', 24],
+    ['3 days', 72],
+    ['1 week', 168],
+    ['Customize', 0],
+  ]);
+
+  DATETIME_FORMAT: string = 'd MMM YYYY h:mm:ss';
+  datePicker: DateFormat = { year: 0, month: 0, day: 0 };
+  timePicker: TimeFormat = { hour: 23, minute: 59 };
+  extendByDatePicker = { minutes: 0, hours: 0, days: 0 };
+
+  onConfirm(): void {
+    if (this.getExtensionTimestamp() < Date.now()) {
+      const extensionTimeString = this.timeZoneService
+      .formatToString(this.getExtensionTimestamp(), this.feedbackSessionTimeZone, this.DATETIME_FORMAT);
+      const currentTimeString = this.timeZoneService
+      .formatToString(Date.now(), this.feedbackSessionTimeZone, this.DATETIME_FORMAT);
+      this.simpleModalService.openConfirmationModal(
+        'Are you sure you want to extend to before the current time?',
+        SimpleModalType.DANGER,
```

Comment:
@damithc Is `DANGER` better, or should we use `WARNING` instead?

Reply:
Probably WARNING is a better match? We usually use DANGER for things such as deleting.
I didn't get the meaning of the `extend to before the current time` phrase though.

Reply:
How about `"Are you sure you wish to set the new deadline to before the current time?"`? Can also add in consequences such as "`Any users affected will have their session closed immediately.`". Though, it is hard to tell what the real consequence is should we allow such behaviour, as it should rightfully invalidate any submissions after the updated deadline even if it is submitted seconds before this action.

Reply:
Perhaps just disallow it? Extensions by definition should be later than the normal deadline. I don't think anybody will ask for a feature that allows the opposite of extensions.

Reply:
Sorry for the vague phrasing. The modal is for the case when there's a pre-existing extension, but the instructor is (accidentally) updating to a time inbetween the original deadline and the current exact time (reducing the extension), effectively closing the session to those who have the extensions. 

The case of opposite extensions is already disallowed (cannot pick with the timepicker) 

I agree with @moziliar's suggestions for the improvement to make it clearer. 

For the mentioned behavior, then would the undefined behavior extend to deletion too? Im thinking then we'll need a warning to instructors to not delete / reduce extensions to those who's submissions will be invalidated by the action. Although this will add additional complexity (most likely in the frontend's validation)

Reply:
Moving the conversation about undefined behavior to the backend PR: https://github.com/TEAMMATES/teammates/pull/11649#issuecomment-1073057610

Reply:
@jayasting98 @samuelfangjw Just adding you guys in for visibility. Also adding in this discussion https://github.com/TEAMMATES/teammates/pull/11649#issuecomment-1073057610. 

The frontend now should have additional validation for:
1. Moving the sessionEndTime to before any extensions (in the sessions edit page), we'll warn of the invalidation of some extensions.
2. Deleting Extensions / Updating pre-existing extensions to before the current date (in the extension page), we'll warn of immediate closure of the session for affected individuals.